crates/ol/stf/src/validation.rs

```diff
@@ -0,0 +1,107 @@
+use strata_crypto::verify_schnorr_sig;
+use strata_ledger_types::{IGlobalState, StateAccessor};
+use strata_ol_chain_types_new::{OLBlock, OLBlockHeader, OLLog, compute_logs_root};
+use strata_params::RollupParams;
+use strata_primitives::CredRule;
+
+use crate::error::BlockValidationError;
+
+/// Block validation before execution. Checks continuity, signature etc.
+pub fn pre_exec_block_validate(
+    block: &OLBlock,
+    prev_header: &OLBlockHeader,
+    params: &RollupParams,
+) -> Result<(), BlockValidationError> {
```

Comment:
This function does a lot of different things which aren't tightly related.  For example,

* block structural consistency checks (which only needs the block header and body)
* header continuity checks (which only needs the block header and the parent header)
* signature checks (which will eventually need the pre-state to know who the proposer is supposed to be)

Also, in the proof we should check that the partial pre-state being provided in the state accessor matches the parent block's state root.  That may be being done here still and I'm just not seeing it.